jenkinsfile-examples/msbuild/Jenkinsfile

```diff
@@ -4,8 +4,11 @@ node {
 
 	stage 'Build'
 		bat 'nuget restore SolutionName.sln'
-		bat "\"${tool 'MSBuild'}\" SolutionName.sln /p:Configuration=Release /p:Platform=\"Any CPU\" /p:ProductVersion=1.0.0.${env.BUILD_NUMBER}"
-
+		bat """
+			setlocal enabledelayedexpansion
+			\"${tool 'MSBuild'}\" SolutionName.sln /p:Configuration=Release /p:Platform=\"Any CPU\" /p:ProductVersion=1.0.0.!BUILD_NUMBER!
```

Comment:
Did you test this?

Even if it works, I'm unsure whether we should do it like this, as it doesn't provide an example of safe Pipeline authoring. There's still the double quote.